### PR TITLE
update unpad doc for slice generality

### PR DESCRIPTION
**Pull request by ntarafdar**

### CODE REVIEW

File: tt_eager/tt_lib/csrc/tt_lib_bindings_tensor_dm_ops.cpp [hunk 1]
```diff
@@ -292,15 +292,15 @@ namespace tt::tt_metal::detail{
...
+            Unpad TT Tensor. Also known as slice.
 
             Returns an output tensor from output tensor start indices ``arg1`` to output tensor end indices ``arg2`` (inclusive) of the input tensor.
 
             .. csv-table::
                 :header: "Argument", "Description", "Data type", "Valid range", "Required"
 
-                "input", "Input tensor", "Tensor", "", "Yes"
-                "output_tensor_start", "Start indices of input tensor", "List[int[4]]", "Must be all 0s", "Yes"
+                "input", "Input tensor", "Tensor", "List[int[4]]", "Yes"
```
Comment (tt-aho) [inline tt_eager/tt_lib/csrc/tt_lib_bindings_tensor_dm_ops.cpp:302]: You should have empty quotes for valid range

**Review by tt-aho — APPROVED** [review]

File: tt_eager/tt_lib/csrc/tt_lib_bindings_tensor_dm_ops.cpp (hunk 1 above)
Comment (tt-aho) [inline tt_eager/tt_lib/csrc/tt_lib_bindings_tensor_dm_ops.cpp:302]: Valid range for input tensor is wrong

File: tt_eager/tt_lib/csrc/tt_lib_bindings_tensor_dm_ops.cpp [hunk 2]
```diff
@@ -292,15 +292,15 @@ namespace tt::tt_metal::detail{
...
+                "output_tensor_start", "Start indices of input tensor", "List[int[4]]", "", "Yes"
```
Comment (tt-aho) [inline tt_eager/tt_lib/csrc/tt_lib_bindings_tensor_dm_ops.cpp:303]: Unpad also supports any tensor rank right? So doesn't need to specifically be 4?

File: tt_eager/tt_lib/csrc/tt_lib_bindings_tensor_dm_ops.cpp (hunk 1 above)
Comment (tarafdarTT) [reply tt_eager/tt_lib/csrc/tt_lib_bindings_tensor_dm_ops.cpp]: oops fixed that

File: tt_eager/tt_lib/csrc/tt_lib_bindings_tensor_dm_ops.cpp (hunk 2 above)
Comment (tarafdarTT) [reply tt_eager/tt_lib/csrc/tt_lib_bindings_tensor_dm_ops.cpp]: ah ya good point! fixed that :)

**Comment by davorchap** [comment]
Should we call this OP "slice" ?

We use it as slice in LLMs ?

**Comment by tarafdarTT** [comment]
@davorchap ya I think we should. I can do that in a follow up PR as that might require changing all references to unpad to use slice, I want this PR to get in ASAP  so @boris-drazic can show this documentation to the syrmia team and they can add sweep tests.

**Comment by davorchap** [comment]
> @davorchap ya I think we should. I can do that in a follow up PR as that might require changing all references of unpad to use slice, I want this PR to get in ASAP  so @boris-drazic can show this documentation to the syrmia team and they can add sweep tests. 

That sounds great

**Pull request merged**